newIDE/app/src/ObjectGroupsList/index.js

```diff
@@ -67,430 +80,524 @@ type Props = {|
-      searchText,
-    });
-    this.displayedGlobalObjectGroupsList = filterGroupsList(
-      globalObjectGroupsList,
-      {
-        searchText,
-      }
+
+    const renderGroupMenuTemplate = React.useCallback(
+      (i18n: I18nType) => (item: TreeViewItem, index: number) =>
+        item.isRoot || item.isPlaceholder
+          ? null
+          : [
+              {
+                label: i18n._(t`Duplicate`),
+                click: () => onDuplicate(item),
+              },
+              { type: 'separator' },
+              {
+                label: i18n._(t`Edit group`),
+                click: () => editItem(item),
+              },
+              { type: 'separator' },
+              {
+                label: i18n._(t`Rename`),
+                click: () => onEditName(item),
+              },
+              {
+                label: i18n._(t`Set as global group`),
+                enabled: !isGroupWithContextGlobal(item),
+                click: () => setAsGlobalGroup(item),
+                visible: canSetAsGlobalGroup !== false,
+              },
+              {
+                label: i18n._(t`Delete`),
+                click: () => onDelete(item),
+              },
+              { type: 'separator' },
+              {
+                label: i18n._(t`Add a new group...`),
+                click: addGroup,
+              },
+            ],
+      [
+        addGroup,
+        onEditName,
+        editItem,
+        onDelete,
+        onDuplicate,
+        canSetAsGlobalGroup,
+        setAsGlobalGroup,
+      ]
     );
-    const fullList = filterGroupsList(
-      [...objectGroupsList, ...globalObjectGroupsList],
-      { searchText }
+
+    const getTreeViewData = React.useCallback(
+      (i18n: I18nType): Array<TreeViewItem> => {
+        const objectGroupsList: GroupWithContextList = enumerateGroups(
+          objectGroups
+        ).map(group => ({ group, global: false }));
+        const globalObjectGroupsList: GroupWithContextList = enumerateGroups(
+          globalObjectGroups
+        ).map(group => ({ group, global: true }));
+
+        const treeViewItems = [
+          {
+            label: i18n._(t`Global Groups`),
+            children:
+              globalObjectGroupsList.length > 0
+                ? globalObjectGroupsList
+                : [
+                    {
+                      label: i18n._(t`There is no global group yet.`),
+                      id: globalGroupsEmptyPlaceholderId,
+                      isPlaceholder: true,
+                    },
+                  ],
+            isRoot: true,
+            id: globalGroupsRootFolderId,
+          },
+          {
+            label: i18n._(t`Scene Groups`),
+            children:
+              objectGroupsList.length > 0
+                ? objectGroupsList
+                : [
+                    {
+                      label: i18n._(t`Start by adding a new group.`),
+                      id: 'scene-empty-placeholder',
```

Comment:
why is this ID not in a variable like the others?

Reply:
Because I didn't need to use it anywhere else, but I will variabalize it